### ISSUE TITLE
Yolo with Tensor RT

### ISSUE THREAD

**Issue by bruchpilot123** [ultralytics/yolov5#7567]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello, 

I tried to use Yolov5 on an Nvidia Jetson with Jetpack 5 together with Tensor RT. I used the following commands: 

`python export.py --weights yolov5s.pt --include engine --imgsz 640 640 --device 0
`
Since TensorRT should be preinstalled with Jetpack5 I did not use the first command from the notebook. Furthermore the first command does not work for me, since it says that there is no version.

During the pyhton export command I get the following error: 

export: data=data/coco128.yaml, weights=['yolov5s.pt'], imgsz=[640, 640], batch_size=1, device=0, half=False, inplace=False, train=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=12, verbose=False, workspace=4, nms=False, agnostic_nms=False, topk_per_class=100, topk_all=100, iou_thres=0.45, conf_thres=0.25, include=['engine']
YOLOv5 🚀 v6.1-161-ge54e758 torch 1.12.0a0+2c916ef.nv22.3 CUDA:0 (Xavier, 31011MiB)

Fusing layers... 
YOLOv5s summary: 213 layers, 7225885 parameters, 0 gradients

PyTorch: starting from yolov5s.pt with output shape (1, 25200, 85) (14.1 MB)
/home/collins/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.1.36ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/home/collins/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.23ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
requirements: nvidia-tensorrt not found and is required by YOLOv5, attempting auto-update...
ERROR: Could not find a version that satisfies the requirement nvidia-tensorrt (from versions: none)
ERROR: No matching distribution found for nvidia-tensorrt
requirements: Command 'pip install 'nvidia-tensorrt' -U --index-url https://pypi.ngc.nvidia.com' returned non-zero exit status 1.

ONNX: starting export with onnx 1.11.0...
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
ONNX: export failure: Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument other in method wrapper__equal)

TensorRT: starting export with TensorRT 8.4.0.9...

TensorRT: export failure: failed to export ONNX file: yolov5s.onnx

Is there something I can do to fix this?

kind regards, 

Robert

**Comment by glenn-jocher** [comment]
@bruchpilot123 commands look correct, just ran them successfully in Colab. We don't have hardware to reproduce on unfortunately, but we have jetson and rpi support coming in the future.

**Comment by zhiqwang** [comment]
@bruchpilot123 ,

Jetson does not support installing tensorrt in this way, see https://docs.nvidia.com/deeplearning/tensorrt/install-guide/index.html#installing-pip for more details. Please use the official jetson documentation https://developer.nvidia.com/embedded/jetpack to install tensorrt.

Also you can see PyTorch's blog https://pytorch.org/blog/running-pytorch-models-on-jetson-nano/ for more details about using YOLOv5 with Jetson.

**Comment by bruchpilot123** [comment]
Sorry, this might be a really stupid questions but I am a beginner. At this time I don`t want to train my own yolov5 with my own dataset, I just want to use the default trained yolov5 for detection. Is this model already optimized for TensorRT? So I don`t have to do anything? Sorry again for the stupid questions but I could not find anything online...

Have a great day :)

**Comment by glenn-jocher** [comment]
@bruchpilot123 export official YOLOv5s model to TRT like this:
```
python export.py --weights yolov5s.pt --include engine
```

See Export tutorial for details:

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️  RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Roboflow for Datasets, Labeling, and Active Learning](https://docs.ultralytics.com/yolov5/tutorials/roboflow_datasets_integration)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TFLite, ONNX, CoreML, TensorRT Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [Architecture Summary](https://docs.ultralytics.com/yolov5/tutorials/architecture_description)&nbsp; ⭐ NEW

Good luck 🍀 and let us know if you have any other questions!
